index.js

```diff
@@ -86,10 +86,15 @@ class Serializer {
     if (typeof i === 'bigint') {
       return i.toString()
     } else if (Number.isInteger(i)) {
-      return this.asNumber(i)
+      return '' + i
     } else {
       /* eslint no-undef: "off" */
-      return this.asNumber(this.parseInteger(i))
+      const integer = this.parseInteger(i)
+      if (isNaN(integer)) {
```

Comment:
Would changing this to `Number.isNaN()` impact performance at all? I know `Global.isNaN()` attempts type coercion.